### ISSUE TITLE
Let typst set 'ViewerPreferences.DisplayDocTitle = true' when exporting to PDF

### ISSUE THREAD

**Issue by JDuffe** [typst/typst#8783]
### Description

Let Typst set the `ViewerPreferences.DisplayDocTitle` dictionary value to `true` in the PDF metadata catalog. This will let PDF Viewers show the document title in the title bar instead of the filename.

### Use Case

I came across this when reading about PDF accessibility. The [matterhorn protocol](https://pdfa.org/resource/the-matterhorn-protocol/), which contains a list of ways, a PDF document can fail the PDF/UA-1 standard, lists (among others) these two points:

> - ViewerPreferences dictionary of the Catalog dictionary does not contain a DisplayDocTitle entry.
> - ViewerPreferences dictionary of the Catalog dictionary contains a DisplayDocTitle entry with a value of false.

There are ways to modify the exported PDF afterwards, for example with the `pikepdf` python package:
```python
import pikepdf

document = pikepdf.open("my-file.pdf", allow_overwriting_input=True)
document.Root.ViewerPreferences = pikepdf.Dictionary(DisplayDocTitle=True)
document.save("my-file.pdf")
```
I think it would be nice if typst did this out of the box, especially because there already is an accessibility guide in the typst documentation.

**Comment by saecki** [comment]
This should be set automatically when exporting a PDF/UA-1  document.
https://github.com/LaurenzV/krilla/blob/fb07499278a2d2524dda38c41f338dec28bd0e48/crates/krilla/src/chunk_container.rs#L263

**Comment by JDuffe** [comment]
Oh, thanks for the hint, that makes sense.
